src/data_hub/api.py

```diff
@@ -796,6 +796,14 @@ def download_database(
     df.to_csv(os.path.join(destination, database_hid + ".csv"))
 
 
+def _replace_with_mapper(item, mapper: dict):
+    """utility function to replace items in a nested list with values generated by a mapper (a dict)."""
+
+    if isinstance(item, list):
+        return [_replace_with_mapper(sub_item, mapper) for sub_item in item]
+    return mapper.get(item, item)
+
+
```

Comment:
this arcane function is to deal with replacing file IDs in columns that can contain one, none, or many files